### DISCUSSION TITLE
One WebButton for 2-Channel Relay

### DISCUSSION THREAD

**Discussion by PetrJakes** [arendst/Tasmota#14280] [Q&A] [answered]
I have 2-Channel WIFI Relay Module configured as an interlock relay.
![image](https://user-images.githubusercontent.com/12581771/148114695-8b33b44e-f3aa-4124-ba19-900d0006732c.png)

Is there a way to configure the web ui to have one webbutton only (instead of two webbuttons) to toggle relays?

**Comment by sfromis** [comment, marked as answer]
As the web UI is just not meant for "daily operation", but only for admin/debug, Tasmota does not provide for much customization. Still, you could use `WebButton` to change the button text to remind you of not clicking there....

**Reply by PetrJakes** [reply]
I just wandered, normally I use MQTT for "daily operations". Thks for your hint anyway.

**Reply by barbudor** [reply]
Interlock mode for 2 relays has 3 possible state : ON/OFF, OFF/ON and OFF/OFF
So you not suitable for 1 single button

**Reply by PetrJakes** [reply]
For my purposes 2 states are OK: ON/OFF, OFF/ON
For this single mode is enough. 
I would like to reverse polarity using following schematic
![image](https://user-images.githubusercontent.com/12581771/148128969-5e161948-49a2-41b4-a49f-afc0559177bc.png)
The OFF/OFF state is not necessary for my case.

**Reply by sfromis** [reply]
Strictly speaking, your case does not matter to what the relay states *can* be. In the web UI, you have one button for each of the two interlocked relays. While turning on the other will turn off the one already on, turning off the one already on will not turn on the other, and both become off. Thus, in the web UI, you cannot control the relays if you remove one of the standard buttons. While you could envision a different web button switching which of the relays is on, that's not how the standard setup works.

**Reply by barbudor** [reply]
Just use a single DPDT relay in that case

**Reply by PetrJakes** [reply]
> Thus, in the web UI, you cannot control the relays if you remove one of the standard buttons.

I was thinking about the scenario where the state of Relay2 is negatively following the state of the Relay1 in the background.
I mean if I toggle the Relay1 ON, Relay2 is going OFF (ideally with some delay) and opposite.

**Reply by PetrJakes** [reply]
> Just use a single DPDT relay in that case

I was not able to find ESP8266 based DPDT relay on the market. Any hint?

**Reply by sfromis** [reply]
While you obviously can envision any setup, including it *always* being either on/off or off/on, that is not how Tasmota works. I'd even say that it is not how it normally "should" work, as most scenarios would include both outputs being off as a very normal situation.

As the Tasmota web UI is not meant as a place to do web design for your specific setup, such tweaks are out of scope for Tasmota. And if you "ideally" want some delay, you should not use interlock, but do an automation yourself.

**Reply by barbudor** [reply]
> > Just use a single DPDT relay in that case
> 
> I was not able to find ESP8266 based DPDT relay on the market. Any hint?

A DPDT relay, a transistor, a resistor and a diode is all you need

**Reply by PetrJakes** [reply]
> A DPDT relay, a transistor, a resistor and a diode is all you need

sure I know this, I was thinking about "ready to use" module, not about any tweaks (even the easy tweaks). But thanks anyway.

**Reply by PetrJakes** [reply]
> such tweaks are out of scope for Tasmota. And if you "ideally" want some delay, you should not use interlock, but do an automation yourself.

I understand and respect it, thks for help. 
PS: while using interlock, there is a delay in the source code already. See https://github.com/arendst/Tasmota/issues/6313

**Reply by barbudor** [reply]
So just to be sure to understand the case, you need only the 2 stable states on/off and off/on but still need a transitionnal off/off state for a certain delay.

While you can't remove buttons from the web GUI, Scripting Language can add and have some routine behind.
So you could at least have a toggle button that in addition can customize any delay you want.
Would that help?

**Reply by PetrJakes** [reply]
> So just to be sure to understand the case, you need only the 2 stable states on/off and off/on but still need a transitionnal off/off state for a certain delay.

I am driving motorized valve (DC motor with the contacts and diodes at the "end" positions) https://docs.google.com/document/d/1zAKR3CXw49x6BhW-twYRRyU00BvHsZGv31fQyLdgyBA/edit?usp=sharing
Now I am driving it with the h-bridge (L293D shield). 
I would like to drive the valve by DPDT relay or by two dependent/interlocking relays.
The delay is necessary to behave nicely to the motor when changing the polarity (when changing rotation of the DC motor)

**Reply by barbudor** [reply]
So the motor is always running in one direction or the other ? It never has a stop state ?

I was wondering if the shutter feature could help you...

**Reply by PetrJakes** [reply]
> So the motor is always running in one direction or the other ? It never has a stop state ?
> 
> I was wondering if the shutter feature could help you...

The motor stops at the end position (limit switch). It is not necessary to switch the power voltage off. It will start to operate opposite direction when reverse voltage occurs.
